apps/full-site-editing/full-site-editing-plugin/wp-template.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/**
+ * Mostly a copy of how the `wp_block` is registered in core
+ *
+ * @return void
+ */
+function fse_register_wp_template() {
+	register_post_type(
+		'wp_template',
+		array(
+			'labels'                => array(
+				'name'                     => _x( 'Templates', 'post type general name' ),
+				'singular_name'            => _x( 'Template', 'post type singular name' ),
+				'menu_name'                => _x( 'Templates', 'admin menu' ),
+				'name_admin_bar'           => _x( 'Template', 'add new on admin bar' ),
+				'add_new'                  => _x( 'Add New', 'Template' ),
+				'add_new_item'             => __( 'Add New Template' ),
+				'new_item'                 => __( 'New Template' ),
+				'edit_item'                => __( 'Edit Template' ),
+				'view_item'                => __( 'View Template' ),
+				'all_items'                => __( 'All Templates' ),
+				'search_items'             => __( 'Search Templates' ),
+				'not_found'                => __( 'No Templates found.' ),
+				'not_found_in_trash'       => __( 'No Templates found in Trash.' ),
+				'filter_items_list'        => __( 'Filter Templates list' ),
+				'items_list_navigation'    => __( 'Templates list navigation' ),
+				'items_list'               => __( 'Templates list' ),
+				'item_published'           => __( 'Template published.' ),
+				'item_published_privately' => __( 'Template published privately.' ),
+				'item_reverted_to_draft'   => __( 'Template reverted to draft.' ),
+				'item_scheduled'           => __( 'Template scheduled.' ),
+				'item_updated'             => __( 'Template updated.' ),
+			),
+			'public'                => true, // temp to support the search endpoint
+			'show_ui'               => true,
+			'show_in_menu'          => true,
+			'rewrite'               => false,
+			'show_in_rest'          => true,
+			'rest_base'             => 'templates',
+			'rest_controller_class' => 'A8C_REST_Templates_Controller',
+			'capability_type'       => 'Template',
+			'capabilities'          => array(
+				// You need to be able to edit posts, in order to read Templates in their raw form.
+				'read'                   => 'edit_posts',
+				// You need to be able to publish posts, in order to create Templates.
+				'create_posts'           => 'customize',
```

Comment:
I'm testing with an administrator user (single site, so no super admin), and I don't have this capability. 🤔 
I can create a template but I cannot publish it: only submit it for review (but to who?).
When the post is saved, the editor get stuck on the "flashing saving icon" because I have no permissions to edit it.
Back in the list, I can see the templates, but cannot do anything with them (they are not linked, nor show any actions).

What about using the same capabilities as `wp_block` instead?

https://github.com/WordPress/WordPress/blob/8e9315f51e4dd60ee77ef7b2d4df0fe8327a9e3b/wp-includes/post.php#L290-L300

Or, like, tighten them up by adding the `_others_` specification?

Reply:
This is very odd. It totally works fine on my local install ( The `customize` capability is straight map for `edit_theme_options`, ([ref](https://github.com/WordPress/WordPress/blob/0344202063022bd9620437ab3c7974e6234f795d/wp-includes/capabilities.php#L498-L499)) which is the default capability for all Customizer settings. 

We can address lockdown a bit more later but this seemed like the right place to start.

@vindl can you replicate?

Reply:
Ok I think the problem is not in the capability, but in the capability _type_:

> The 'capability_type' parameter is used as a base to construct capabilities unless they are explicitly set with the 'capabilities' parameter. It seems that `map_meta_cap` needs to be set to false or null, to make this work.

https://codex.wordpress.org/Function_Reference/register_post_type#capability_type

For some reasons I can't understand, this is what is working for me:
```php
//'capability_type'       => 'template',
'capabilities'          => array(
	// You need to be able to edit posts, in order to read templates in their raw form.
	'read'                   => 'edit_posts',
	// You need to be able to customize, in order to create templates.
	'create_posts'           => 'edit_theme_options',
	'edit_posts'             => 'edit_theme_options',
	'edit_published_posts'   => 'edit_theme_options',
	'delete_published_posts' => 'edit_theme_options',
	'edit_others_posts'      => 'edit_theme_options',
	'delete_others_posts'    => 'edit_theme_options',
),
'map_meta_cap'          => true,
```

Basically, _without_ `capability_type` and _with_ `map_meta_cap`, which though doesn't map `customize` as I would have expected. 😅